File: GenesisAndroid/src/main/java/com/emerchantpay/gateway/genesisandroid/api/interfaces/customerinfo/CustomerInfoAttributes.java
package com.emerchantpay.gateway.genesisandroid.api.interfaces.customerinfo;

import com.emerchantpay.gateway.genesisandroid.api.util.RequestBuilder;

public interface CustomerInfoAttributes {

    RequestBuilder requestBuilder = new RequestBuilder("");

    // Customer Info Params
    default CustomerInfoAttributes setCustomerEmail(String customerEmail) {
            requestBuilder.addElement("customer_email", customerEmail);
            return this;
    }

    default CustomerInfoAttributes setCustomerPhone(String customerPhone) {
        requestBuilder.addElement("customer_phone", customerPhone);
        return this;
    }

    default RequestBuilder buildCustomerInfoParams() {
        return requestBuilder;
    }
}

File: GenesisAndroid/src/main/java/com/emerchantpay/gateway/genesisandroid/api/interfaces/customerinfo/CustomerInfoAttributes.kt
package com.emerchantpay.gateway.genesisandroid.api.interfaces.customerinfo

import com.emerchantpay.gateway.genesisandroid.api.util.RequestBuilder

interface CustomerInfoAttributes {

    fun setCustomerEmail(customerEmail: String): CustomerInfoAttributes {
        requestBuilder.addElement("customer_email", customerEmail)
        return this
    }

    fun setCustomerPhone(customerPhone: String): CustomerInfoAttributes {
        requestBuilder.addElement("customer_phone", customerPhone)
        return this
    }

    fun buildCustomerInfoParams(): RequestBuilder {
        return requestBuilder
    }

    companion object {
        val requestBuilder = RequestBuilder("")
    }
}